src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -242,7 +247,7 @@ protected HandleReferencesBase(Index index, DocumentsStorage documentsStorage, I
                                                 if (CanContinueReferenceBatch() == false)
                                                 {
                                                     // updating the last reference state in order to continue from the place we left off
-                                                    referenceState = new ReferencesState.ReferenceState(referencedItem.Key, referencedItem.Etag, current.LowerSourceDocumentId ?? current.Id, lastIndexedParentEtag);
```

Comment:
`current.LowerSourceDocumentId ?? current.Id` worked because,
for `Documents` indexes, `LowerSourceDocumentId` is always `null` and we fill it only for `Counters` and `TimeSeries` indexes.